### PR TITLE
Added API view to get best matching volunteers for helprequest

### PR DESCRIPTION
**Pull request by stavlocker**

### CODE REVIEW

File: api/views.py [hunk 1]
```diff
@@ -134,6 +135,39 @@ class VolunteersViewSet(mixins.ListModelMixin, viewsets.GenericViewSet):
...
+            )
+        try:
+            helprequest_id = int(helprequest_id)
+        except ValueError:
+            return Response(
+                {'helprequest_id': f'No help request with ID {helprequest_id} found.'},
+                status=status.HTTP_400_BAD_REQUEST
+            )
+        helprequests = HelpRequest.objects.filter(pk=helprequest_id)
+        if helprequests.count() == 0:
```
Comment (astar26) [inline api/views.py:159]: Use exists()

File: api/views.py [hunk 2]
```diff
@@ -134,6 +135,39 @@ class VolunteersViewSet(mixins.ListModelMixin, viewsets.GenericViewSet):
...
+            return Response(
+                {'helprequest_id': f'No help request with ID {helprequest_id} found.'},
+                status=status.HTTP_400_BAD_REQUEST
+            )
+
+        coords = (helprequests[0].location_latitude, helprequests[0].location_longitude)
```
Comment (astar26) [inline api/views.py:165]: Rule of thunb - use the first() method, it absorbs some possible errors.

File: api/views.py [hunk 3]
```diff
@@ -134,6 +135,39 @@ class VolunteersViewSet(mixins.ListModelMixin, viewsets.GenericViewSet):
...
+        queryset = Volunteer.objects.all_by_score(coords)[:type(self).BEST_MATCH_VOLUNTEERS_LIMIT]
```
Comment (astar26) [inline api/views.py:166]: is the type(self) necessary?

File: api/views.py (hunk 1 above)
Comment (stavlocker) [reply api/views.py:159]: resolved

File: api/views.py [hunk 4]
```diff
@@ -134,6 +135,39 @@ class VolunteersViewSet(mixins.ListModelMixin, viewsets.GenericViewSet):
...
+            )
+        try:
+            helprequest_id = int(helprequest_id)
+        except ValueError:
+            return Response(
+                {'helprequest_id': f'No help request with ID {helprequest_id} found.'},
+                status=status.HTTP_400_BAD_REQUEST
+            )
+        helprequest = HelpRequest.objects.filter(pk=helprequest_id).first()
+        if helprequest is None:
```
Comment (astar26) [inline api/views.py:159]: Code repetition. Maybe convert to exception and catch both?

**Comment by stavlocker** [comment]
@astar26 Fixed comments

**Pull request merged**